src/cosalib/cmdlib.py

```diff
@@ -249,10 +249,19 @@ def import_ostree_commit(repo, buildpath, buildmeta, force=False):
         return
 
     # extract in a new tmpdir inside the repo itself so we can still hardlink
-    with tempfile.TemporaryDirectory(dir=repo) as d:
-        subprocess.check_call(['tar', '-C', d, '-xf', tarfile])
-        subprocess.check_call(['ostree', 'pull-local', '--repo', repo,
-                               d, commit])
+    if tarfile.endswith('.tar'):
+        with tempfile.TemporaryDirectory(dir=repo) as d:
+            subprocess.check_call(['tar', '-C', d, '-xf', tarfile])
+            subprocess.check_call(['ostree', 'pull-local', '--repo', repo,
+                                   d, commit])
+    elif tarfile.endswith('.ociarchive'):
+        # We do this in two stages, because right now ex-container only writes to
+        # non-archive repos.  Also, we need sudo to write to `repo-build`, though
+        # it might be good to change this by default.
+        build_repo = os.path.join(repo, '../../cache/repo-build')
+        subprocess.check_call(['sudo', 'rpm-ostree', 'ex-container', 'import', '--repo', build_repo,
+                               '--write-ref', buildmeta['buildid'], 'oci-archive:' + tarfile])
+        subprocess.check_call(['sudo', 'ostree', f'--repo={repo}', 'pull-local', build_repo, buildmeta['buildid']])
```

Comment:
Should we `chown -R` the whole repo back here? (See e.g. `runcompose_tree`).

Reply:
Urg, good catch.  Actually I'd only tested this whole thing in the privileged path.  I added some (as yet untested) code to handle unprivileged.

The more I think about this though, the more I feel like the whole `tmp/repo` thing was a mistake.  Basically, if we're just going to eat the write amplification of making a big tarball (and qcow2) as a general rule, then keeping a `tmp/repo` is mostly pointless.  All we need is the commit object to do things like rpmdb diffs.

And so e.g. we should then switch `create_disk.sh` to run e.g. `rpm-ostree ex-container import` instead.  Almost everything else operating on `tmp/repo` just wants the commit object.

Reply:
Hmm, I think all the places which only need the commit object already do that optimization, no? If not, then yeah we shouldn't unpack for that. Everything else though does need the repo on disk to e.g. copy files out/inspect/build new trees and it seems like we'd be wasting an important optimization if we just nuked the repo instead of leaving it around after a build.

Reply:
For `cosa buildfast` at least, I could imagine teaching `ostree commit` to take e.g. `--base path/to/my.ociarchive` in the future.

Reply:
>  it seems like we'd be wasting an important optimization if we just nuked the repo instead of leaving it around after a build.

The problem we hit is that for the disk image we hit issues with qemu reading xattrs over 9p which meant we can't directly read `cache/repo-build`.  That might be fixed by virtiofs.

I'm not proposing deleting `cache/repo-build` but...leaving it as a cache just for rpm-ostree, not for anything else.